core/chains/evm/client/simulated_backend_client.go

```diff
@@ -654,32 +475,168 @@ func (c *SimulatedBackendClient) Commit() common.Hash {
 	return c.b.Commit()
 }
 
-func toCallMsg(params map[string]interface{}) ethereum.CallMsg {
-	var callMsg ethereum.CallMsg
+func (c *SimulatedBackendClient) IsL2() bool {
+	return false
+}
 
-	switch to := params["to"].(type) {
-	case string:
-		toAddr := common.HexToAddress(to)
-		callMsg.To = &toAddr
-	case common.Address:
-		callMsg.To = &to
-	case *common.Address:
-		callMsg.To = to
+func (c *SimulatedBackendClient) fetchHeader(ctx context.Context, blockNumOrTag string) (*types.Header, error) {
+	switch blockNumOrTag {
+	case rpc.SafeBlockNumber.String():
+		return c.b.Blockchain().CurrentSafeBlock(), nil
+	case rpc.LatestBlockNumber.String():
+		return c.b.Blockchain().CurrentHeader(), nil
+	case rpc.FinalizedBlockNumber.String():
```

Comment:
How about comparing as `BlockNumber`?
```suggestion
	switch rpc.BlockNumber(blockNumOrTag) {
	case rpc.SafeBlockNumber:
		return c.b.Blockchain().CurrentSafeBlock(), nil
	case rpc.LatestBlockNumber:
		return c.b.Blockchain().CurrentHeader(), nil
	case rpc.FinalizedBlockNumber:
```

Reply:
To do that, I would need to cast to an int64, right?

`type BlockNumber int64`

In the call, it is being passed as a string. This function seems to just be just running through the list of string options until finally trying to cast it to `*big.Int`.

Reply:
Oh yeah, nvm. I misread.